examples_tests/39.DenoiserTonemapper/main.cpp

```diff
@@ -630,83 +629,163 @@ void main()
 			};
 
 			auto& color = getImageAssetAndItsChannelName(color_image_bundle);
-			auto& albedo = getImageAssetAndItsChannelName(albedo_image_bundle);
-			auto& normal = getImageAssetAndItsChannelName(normal_image_bundle);
+			decltype(color)& albedo = getImageAssetAndItsChannelName(albedo_image_bundle);
+			decltype(color)& normal = getImageAssetAndItsChannelName(normal_image_bundle);
 
-			/*
-					TODO: actually why don't we want exr inputs?
-				if (image->getCreationParameters().format!=irrFmtRequired)
+			auto convertImageToRequiredFmt = [&](core::smart_refctd_ptr<ICPUImage> image)
+			{
+				using CONVERSION_FILTER = CConvertFormatImageFilter<>;
+
+				core::smart_refctd_ptr<ICPUImage> newConvertedImage;
 				{
-					CConvertFormatImageFilter<>::state_type state;
-					state.extentLayerCount =
-					state.inImage = image;
+					auto referenceImageParams = image->getCreationParameters();
+					auto referenceBuffer = image->getBuffer();
+					auto referenceRegions = image->getRegions();
+					auto referenceRegion = referenceRegions.begin();
+					const auto newTexelOrBlockByteSize = asset::getTexelOrBlockBytesize(irrFmtRequired);
+
+					auto newImageParams = referenceImageParams;
+					auto newCpuBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(referenceRegion->getExtent().width * referenceRegion->getExtent().height * referenceRegion->getExtent().depth * newTexelOrBlockByteSize);
+					auto newRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(referenceRegions.size());
+
+					*newRegions->begin() = *referenceRegion;
+
+					newImageParams.format = irrFmtRequired;
+					newConvertedImage = ICPUImage::create(std::move(newImageParams));
+					newConvertedImage->setBufferAndRegions(std::move(newCpuBuffer), newRegions);
+
+					CONVERSION_FILTER convertFilter;
+					CONVERSION_FILTER::state_type state;
+
+					state.inImage = image.get();
+					state.outImage = newConvertedImage.get();
+					state.inOffset = { 0, 0, 0 };
+					state.inBaseLayer = 0;
+					state.outOffset = { 0, 0, 0 };
+					state.outBaseLayer = 0;
+
+					auto region = newConvertedImage->getRegions().begin();
+
+					state.extent = region->getExtent();
+					state.layerCount = region->imageSubresource.layerCount;
+					state.inMipLevel = region->imageSubresource.mipLevel;
+					state.outMipLevel = region->imageSubresource.mipLevel;
+
+					if (!convertFilter.execute(&state))
+						os::Printer::log("WARNING (" + std::to_string(__LINE__) + " line): Something went wrong while converting the image!", ELL_WARNING);
```

Comment:
what's the deal with this?

Reply:
you have started something with the filter and commented it, so I did it for you if you were to use it